src/Pages/RoomList/FilterDropDown/FilterDropDown.js

```diff
@@ -0,0 +1,178 @@
+import React, { useState, useRef, useEffect } from "react";
+import { useHistory } from "react-router-dom";
+import RoomTypeCheckItem from "../FilterItems/CheckItem/RoomTypeCheckItem";
+import styled from "styled-components";
+import { flexCenter, flexAlignCenter } from "../../../styles/Theme";
+
+const FilterDropDown = ({
+  handleExit,
+  setQueryRoom,
+  historyPush,
+  fetchFilteredData,
+}) => {
+  const history = useHistory();
+  const popUpRef = useRef();
+  const backgroundRef = useRef();
+  const [roomTypeFilters, setRoomTypeFilters] = useState([]);
+  const [roomTypeFilteredResult, setRoomTypeFilteredResult] = useState([]);
+
+  useEffect(() => {
+    fetch("/data/RoomList/roomTypeFilters.json")
+      .then((res) => res.json())
+      .then((data) => {
+        const newData = data.roomTypeFilters.map((filter) => {
+          return { ...filter, isChecked: false };
+        });
+        setRoomTypeFilters(newData);
+      });
+  }, []);
+
+  const handleRoomTypeCheck = (id) => {
+    const changedStatus = roomTypeFilters.map((filter) => {
+      if (filter.id === id) {
+        filter.isChecked = !filter.isChecked;
+      }
+      return filter;
```

Comment:
```suggestion
    const changedStatus = roomTypeFilters.map((filter) => {
      if (filter.id === id) return !filter
      return filter
```
- for문이 아닌 map 함수를 사용하는 이유가 무엇일까요?
- 전체적으로 수정해주시기 바랍니다! filter 기능 구현하면서 array method 사용하신 부분 확인해주세요.

Reply:
필터링된 리스트를 가지고 쿼리스트링을 만들기 위해 + 드롭다운 메뉴를 저장하지 않고 나갔을 때와 저장한 후를 구분하기 위해서 그렇게 짰던 것 같습니다. 백엔드에서 필터 목록은 전달해주지 못해 제가 임시로 만들어 놓은 목데이터에서 로직을 추가하다 보니 그렇게 됐던 것 같습니다. 
그런데 이부분이 갑자기 문제가 생겨서 멘탈이 나갈 것 같습니다ㅠㅠ 이런 로직을 짠 것이 원인이 되는지는 모르겠는데 필터 팝업 이벤트가 한 번 늦게 발생하는 문제가 있고, 체크 후 저장을 눌러 필터가 된 상태에서 다시 팝업을 열면 체크가 되어 있는 상태로 남아 있도록 짜고 잘 작동하는 것을 확인했거든요! 그런데 어제부터 팝업을 열 때마다 이 state가 리셋이 됩니다. 팝업 여는 함수를 찾아봐도 별다른 문제가 보이지 않는데 roomTypeFilteredResult를 콘솔에 찍어 보면 팝업 열 때마다 isChecked: true로 표시되었던 항목도 false로 다 리셋이 되네요. 으아아

Reply:
- 코드에 누락된 부분이 있는데 다시 확인해보세요.
```suggestion
    const changedStatus = roomTypeFilters.map((filter) => {
      if (filter.id === id) return {...filter, isChecked: !filter.isChecked}
      return filter
```
- 실제 이벤트에 비해 팝업 한번 늦게 실행된다면 setState 비동기 문제일 확률이 높아보입니다.